From andi@firstfloor.org  Thu Jul  3 16:21:46 2008
From: Andi Kleen <andi@firstfloor.org>
Date: Tue,  1 Jul 2008 18:48:43 +0200 (CEST)
Subject: sysdev: Convert the x86 mce tolerant sysdev attribute to generic attribute
To: greg@kroah.com, linux-kernel@vger.kernel.org
Message-ID: <20080701164843.EA94A1B4314@basil.firstfloor.org>


From: Andi Kleen <andi@firstfloor.org>

Use the new generic int attribute accessors for the x86 mce tolerant
attribute. Simple example to illustrate the new macros.

There are much more places all over the tree that could be converted
like this.

Signed-off-by: Andi Kleen <ak@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 arch/x86/kernel/cpu/mcheck/mce_64.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/arch/x86/kernel/cpu/mcheck/mce_64.c
+++ b/arch/x86/kernel/cpu/mcheck/mce_64.c
@@ -800,12 +800,12 @@ static ssize_t set_trigger(struct sys_de
 }
 
 static SYSDEV_ATTR(trigger, 0644, show_trigger, set_trigger);
-ACCESSOR(tolerant,tolerant,)
+static SYSDEV_INT_ATTR(tolerant, 0644, tolerant);
 ACCESSOR(check_interval,check_interval,mce_restart())
 static struct sysdev_attribute *mce_attributes[] = {
 	&attr_bank0ctl, &attr_bank1ctl, &attr_bank2ctl,
 	&attr_bank3ctl, &attr_bank4ctl, &attr_bank5ctl,
-	&attr_tolerant, &attr_check_interval, &attr_trigger,
+	&attr_tolerant.attr, &attr_check_interval, &attr_trigger,
 	NULL
 };
 
